### PR TITLE
Added internal paging

### PR DESCRIPTION
**Pull request by bensgroi**
Added the internal paging feature, so sorting now works on the paging demo.
Disabled sorting on the virtual paging demo, since built-in sorting is incompatible at this time.

Works to include fix for #33

### CODE REVIEW

**Comment by jonshaffer** [comment]
This looks good. Can you add some a couple unit tests and if you're feeling ambitious a protractor test for sorting+paging on the demo?

File: demos/paging.html [hunk 1]
```diff
@@ -1,102 +1,88 @@
...
-              set.forEach(function(r, i){
-                var idx = i + (offset * size);
-                $scope.data[idx] = r;
-              });
-
-              console.log('paging!', offset, size)
-            });
-          }, 200)
-          };
+<head>
```
Comment (jonshaffer) [inline demos/paging.html]: it looks like you tabbed <head at the same level of <html?

File: demos/virtual-paging.html [hunk 2]
```diff
@@ -70,9 +70,9 @@
             headerHeight: 50,
             footerHeight: 50,
             columns: [
-              { name: "Name", prop: "name" },
-              { name: "Gender", prop: "gender" },
-              { name: "Company", prop: "company" }
+              { name: "Name", prop: "name", sortable: false },
```
Comment (jonshaffer) [inline demos/virtual-paging.html]: looks like we could use a global 'sortable'!

File: src/components/body/BodyController.js [hunk 3]
```diff
@@ -137,13 +155,17 @@ export default class BodyController {
...
         if (this.options.paging.externalPaging) {
+          // We're using external paging
           let idxs = this.getFirstLastIndexes(),
               idx = idxs.first;
 
           this.tempRows.splice(0, this.tempRows.length);
           while (idx < idxs.last) {
             this.tempRows.push(rows[idx++])
           }
+        } else if (this.options.paging.size) {
```
Comment (jonshaffer) [inline src/components/body/BodyController.js]: It looks like this is nested in a this.options.paging.size already, which would prevent the next else statement from ever firing. Looks safe to remove the if condition and the else below it, but you may know more about this area than me now :)

That being said - do we know the relation of the externalPaging logic to all of that paging that happens in virtual paging?

File: demos/paging.html (hunk 1 above)
Comment (bensgroi) [inline demos/paging.html]: Yeah, VS Code doesn't do a great job at auto-formatting. I'll fix.

File: demos/virtual-paging.html (hunk 2 above)
Comment (bensgroi) [inline demos/virtual-paging.html]: True. Added issue #53

**Pull request merged**